WORKSPACE

```diff
@@ -107,6 +107,18 @@ npm_translate_lock(
             "requires-network",
         ],
     },
+    lifecycle_hooks_sandboxed = [
+        # If @kubernetes/client-node is not sandboxed, will fail with
+        # ```
+        # src/azure_auth.ts(97,43): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # src/azure_auth.ts(98,34): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # src/gcp_auth.ts(93,43): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # src/gcp_auth.ts(94,34): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # ```
+        # since a `jsonpath-plus@7.2.0` that is newer then the transitive dep `jsonpath-plus@0.19.0` is found outside of the sandbox that
+        # includes typings that don't match the 0.19.0 "any" usage.
+        "@kubernetes/client-node",
```

Comment:
it feels like it's going to be a forever project for our community to discover all the brokenness and add them to their WORKSPACE files. I'm not sure what to do about it.

Reply:
3p deps are one of the harder parts and lifecycle hooks will be one of those sharp edges. The `@kubernetes/client-node` package is the first one I've seen the requires `prepare` and actually runs typescript in the `prepare` step to build a `dist` folder. This is an example of one that could use a BUILD file.

Reply:
I keep wishing for us to have some registry of known ones that ppl need, maybe even fetch that as a default input to the rule so it gets easier over time. Like that repo that yarn and pnpm co-maintain that has common hoisting.

But it would have to be versioned (some later `@kubernetes/client-node` might fix it). Seems intractable.

Maybe the answer is just to keep improving error messages so that users discover the right WORKSPACE settings.

Another thought: instead of these having to go in the root workspace it would be nice to put them in package.json with the pnpm hoist settings.

Thoughts for another day